.github/workflows/ca-clone-sequential-test.yml

```diff
@@ -984,6 +984,481 @@ jobs:
 
           diff expected output
 
+      ####################################################################################################
+      # Switch cert request ID generator to legacy2 and verify if serials
+      # have gaps when range is updated
+      #
+      # It should work like the legacy but with correct range.
+      - name: Switch primary to legacy2
+        run: |
+          docker exec primary pki-server stop
+          docker exec primary pki-server ca-range-generator-update --type request legacy2
+          docker exec primary pki-server ca-range-generator-update --type cert legacy2
+          docker exec primary pki-server start --wait
+
+      - name: Check cert range objects
+        run: |
+          tests/ca/bin/ca-cert-range-objects.sh primaryds | tee output
+
+          # cert ranges should remain the same
+          cat > expected << EOF
+          SecurePort: 8443
+          beginRange: 31
+          endRange: 48
+          host: secondary.example.com
+
+          SecurePort: 8443
+          beginRange: 19
+          endRange: 48
+          host: primary.example.com
+
+          EOF
```

Comment:
To my understanding the `beginRange` and `endRange` attributes are always in decimal and we're not changing it in this PR:
https://github.com/dogtagpki/pki/blob/master/base/server/src/main/java/com/netscape/cmscore/dbs/Repository.java#L574-L578

The values are derived from `nextRange` which was parsed as decimal:
https://github.com/dogtagpki/pki/blob/master/base/server/src/main/java/com/netscape/cmscore/dbs/Repository.java#L543-L544

Reply:
Yes, they are written in decimal and derived from nextRange. However, nextRange is read as decimal when assigned to the range but it is read as hex when assigned in nextMinSerialNumber:
https://github.com/dogtagpki/pki/blob/a76eb4b8b1d3be26de69eae36ce0fa560d3b7e72/base/server/src/main/java/com/netscape/cmscore/dbs/Repository.java#L701

Therefore the value written in range match the value written in CS.cfg but the real value is different. With this patch the value will really match after  the conversion.

Reply:
Yeah, that makes it really complicated since we cannot really rely on the values written in the `CS.cfg` or DS so there are more possibilities of problems. I need to look at the code to see how `mNextMinSerialNo` is used.

So in this step the range size grows from 18 (which is the size specified during installation) to 30, and it temporarily overlaps the other server's range. Where does the 30 come from?

Reply:
There is no real overlap because the one updated is in decimal while the other is in hex. If we took the range of the primary as example here. Initially the range was from 13 to 30 (30 = 13 + increment in dec - 1). The following range start from 31 and the begin is the real one. The gap is generated because in the CS.cfg the end is not 30 but 13 + incerement in hex - 1. = 24 and the next start from 31.

BTW, I think I found a possible conflict and I am updating the code.